js/src/components/TaxonomyPicker.js

```diff
@@ -0,0 +1,89 @@
+/* External dependencies */
+import React from "react";
+import PropTypes from "prop-types";
+
+/**
+ * A select box for selecting a taxonomy.
+ *
+ * Also calls `onChange` prop when the currently selected options is no longer available.
+ */
+class TaxonomyPicker extends React.Component {
+	/**
+	 * Checks if the terms have changes and calls the appropriate handler.
+	 *
+	 * @param {Object} prevProps The previous props.
+	 *
+	 * @returns {void}
+	 */
+	componentDidUpdate( prevProps ) {
+		const { terms } = this.props;
+		if ( terms.length > 0 && terms !== prevProps.terms ) {
+			this.handleTermsChange();
+		}
+	}
+
+	/**
+	 * Checks if the current value still has a corresponding option, and if not changes
+	 * the value to the first term's id.
+	 *
+	 * @returns {void}
+	 */
+	handleTermsChange() {
+		const { terms, value } = this.props;
+		const selectedTerm = terms.find( term => {
+			return term.id === value;
+		} );
+		if ( ! selectedTerm ) {
+			this.props.onChange( terms[ 0 ].id );
+		}
+	}
+
+	/**
+	 * Renders the TaxonomyPicker component.
+	 *
+	 * @returns {ReactElement} The rendered component.
+	 */
+	render() {
+		const {
+			value,
+			id,
+			terms,
+			onChange,
+		} = this.props;
+
+		return (
+			<select
+				disabled={ terms.length < 2 }
```

Comment:
Wouldn't it be better to hide the dropdown if there is only one category selected? Currently, the classic editor doesn't show 'make primary' when there is only one (or none) category selected.